core/src/main/java/org/eclipse/hono/connection/impl/ConnectionFactoryImpl.java

```diff
@@ -82,6 +84,11 @@ public String getName() {
         return config.getName();
     }
 
+    @Override
+    public String getContainerId() {
+        return String.format("%s-%s", getName(), containerIdUuidPart);
+    }
+
```

Comment:
FMPOV each connection created by this factory should have its own unique container ID, e.g.
```
private String getContainerId() {
  return String.format("%s-%s-%s", config.getName(), config.getServerRole(), UUID.randomUUID())
}
```

Reply:
Letting `ConnectionFactoryImpl.connect()` assign a new random container ID to each newly created `ProtonConnection` will prevent us from using it as the `adapterInstanceId` though. Otherwise a reconnect on the `HonoConnection` used by the `ProtocolAdapterCommandConsumerFactory` would cause the `adapterInstanceId` to be different from the container id.

Sticking to the plan of keeping `adapterInstanceId` and container id the same would mean having the container id stored in the `HonoConnectionImpl` instance and setting that same id on every `ProtonConnection` instance that gets created via the `HonoConnectionImpl` instance.
I think that is a valid approach - the `HonoConnection` only has one open `ProtonConnection` at a time, so there wouldn't be 2 open `ProtonConnection`s with the same container id.

To do this, I would add a `ConnectionFactory.connect` method with an extra container id parameter.

Reply:
I've committed the aforementioned changes. I've also removed `ConnectionFactory.getName()`  - its javadoc stated that the name is used in the container id - that is not _necessarily_ the case anymore because the container id is now provided as an extra param to `ConnectionFactory.connect()`.